gcc/config/aarch64/aarch64-coff.h

```diff
@@ -54,19 +58,20 @@
     }
 #endif
 
-/* Output a local common block.  /bin/as can't do this, so hack a
-   `.space' into the bss segment.  Note that this is *bad* practice,
-   which is guaranteed NOT to work since it doesn't define STATIC
-   COMMON space but merely STATIC BSS space.  */
-#ifndef ASM_OUTPUT_ALIGNED_LOCAL
-# define ASM_OUTPUT_ALIGNED_LOCAL(STREAM, NAME, SIZE, ALIGN)		\
-    {									\
-      switch_to_section (bss_section);					\
-      ASM_OUTPUT_ALIGN (STREAM, floor_log2 (ALIGN / BITS_PER_UNIT));	\
-      ASM_OUTPUT_LABEL (STREAM, NAME);					\
-      fprintf (STREAM, "\t.space\t%d\n", (int)(SIZE));			\
-    }
-#endif
+#define ASM_OUTPUT_LOCAL(FILE, NAME, SIZE, ROUNDED)  \
+( fputs (".lcomm ", (FILE)),			\
+  assemble_name ((FILE), (NAME)),		\
+  fprintf ((FILE), ",%u\n", (int)(ROUNDED)))
+
+#define ASM_OUTPUT_ALIGNED_LOCAL(FILE, NAME, SIZE, ALIGNMENT)  \
+  { \
+    unsigned HOST_WIDE_INT rounded = MAX((SIZE), 1); \
+    unsigned HOST_WIDE_INT alignment = MAX((ALIGNMENT), BIGGEST_ALIGNMENT); \
+    rounded += (alignment / BITS_PER_UNIT) - 1; \
+    rounded = (rounded / (alignment / BITS_PER_UNIT) \
```

Comment:
Some of the other targets outputs `.align` directive using `ASM_OUTPUT_ALIGN`. Why it cannot be used here? Is there any other target doing it the same way as here?

Reply:
I suspect that BIGGEST_ALIGNMENT on AArch64 is smaller than on ix86. This is why MAX((ALIGNMENT), BIGGEST_ALIGNMENT) is needed.
No, it is custom alignment for AArch64 based on default implementation.

Reply:
I am trying to understand what is the difference between AArch64 ELF and AArch64 COFF when handling the code here [gcc-woarm64/gcc/varasm.cc](https://github.com/Windows-on-ARM-Experiments/gcc-woarm64/blob/woarm64/gcc/varasm.cc#L2137). If I am interpreting the code right, neither `ASM_OUTPUT_ALIGNED_DECL_LOCAL`, `ASM_OUTPUT_ALIGNED_LOCAL`, or `ASM_OUTPUT_LOCAL` seems to be defined for AArch64 ELF, which effectively just returns `false` for this function while for AArch64 COFF with this PR's changes will emit some code and return `true`. Why is this different handling needed?

Reply:
I do not know why it is not needed in ELF. In COFF it is definitely needed.

Reply:
So what this PR effectively changes in temms of the assembly code generated?

Reply:
It handles an alignment when it is higher than default BIGGEST_ALIGNMENT which is 16 and emit .lcomm with the the calculated alignment.

Reply:
OK, thank you for the clarification.